unity-ggjj/Assets/Scripts/TextDecoder/NarrativeScriptPlaylist.cs

```
@@ -0,0 +1,78 @@
+using System.Collections.Generic;
+using UnityEngine;
+using UnityEngine.Serialization;
+
+[RequireComponent(typeof(SceneLoader))]
+public class NarrativeScriptPlaylist : MonoBehaviour
+{
+    [field: Tooltip("List of narrative scripts to be played in order")]
+    [field: SerializeField] public List<NarrativeScript> NarrativeScripts { get; private set; }
+    
+    [field: Tooltip("List of narrative scripts to be used when taking penalties")]
+    [field: SerializeField] public List<NarrativeScript> FailureScripts { get; private set; }
+    
+    [field: Tooltip("A narrative script to be played on game over")]
+    [field: SerializeField] public NarrativeScript GameOverScript { get; private set; }
+    
+    [Tooltip("Write the name of the next scene to load here.")]
+    [SerializeField] private string _nextSceneName;
```

Nit: I'm unsure if this field and it's associated logic are still relevant after this PR. This field's value is used if we "run out of lines" to execute and I would make the argument, this is something we should throw and abort execution for instead; it's actually an unexpected state:

Generally, the game's flow is planned to be completely defined via the `.ink`-scripts; thereby running out of lines _shouldn't_ really happen. If we encounter the final dialogue for the current state of the game, we should instead transition to credits (or the main menu for the time being) and do so via the `&SCENE:`-action, to give that power to writers instead of limiting it to people working in the editor only. Doing so, the Unity scene is left before "no lines" are left inside the narrative script.

Does that approach make sense to you too? Maybe there's also another case where this could prove useful I've missed, so if I did, do let me know!

---

I agree, and it also doesn't make sense for the one scene for all scripts approach we're going for in this PR. Are you suggesting having, for example, a &LOAD_SCRIPT action at the end of every narrative script? This would require a way of either reloading a scene and injecting the script to play (which I think needs to be done at some point anyway), or destroying the assets for the current scene and loading new ones (or not destroying assets that need to be used in the new script if we want to be SUPER efficient). Maybe this should be its own issue and PR as I'm not sure how much time I will be able to dedicate at the moment with university.

---

Just to focus on the field first: I'd say let's remove it for now as (I believe) it's completely unused, since we don't have any sort of scene progression yet, right?  
If that works for you, feel free to just make that change and mark this thread as resolved or let me know if you already have an alternative in mind you still want to make a part of this branch.

That said: I think splitting this up into a separate issue is definitely a good call and that `&LOAD_SCRIPT`-actions fit perfectly into something I was already thinking about:
[I've moved the "scene and script progression" aspect of it to a discussion](https://github.com/Studio-Lovelies/GG-JointJustice-Unity/discussions/170), so we can focus this thread on the field specifically, but it's a super important topic, and I didn't want it to get drowned out and forgotten as "just another PR nitpick"! I hope that's okay for you too!

---

Removed scene loading from NarrativeScriptPlaylist